File: app/src/main/java/ru/magflayer/spectrum/presentation/pages/splash/SplashPresenter.java
package ru.magflayer.spectrum.presentation.pages.splash;

import com.arellomobile.mvp.InjectViewState;

import javax.inject.Inject;

import ru.magflayer.spectrum.domain.injection.InjectorManager;
import ru.magflayer.spectrum.domain.interactor.ColorInfoInteractor;
import ru.magflayer.spectrum.presentation.common.android.navigation.router.GlobalRouter;
import ru.magflayer.spectrum.presentation.common.mvp.BasePresenter;

@InjectViewState
public class SplashPresenter extends BasePresenter<SplashView> {

    @Inject
    GlobalRouter globalRouter;
    @Inject
    ColorInfoInteractor colorInfoInteractor;

    void openMainPage() {
        execute(colorInfoInteractor.uploadColorInfo(),
                colorInfoState -> logger.debug("Color info loaded: {}", colorInfoState),
                error -> {
                    logger.warn("Error while uploaded color info: ", error);
                    globalRouter.startMain();
                },
                () -> globalRouter.startMain());
    }

    @Override
    protected void inject() {
        InjectorManager.getAppComponent().inject(this);
    }
}

File: app/src/main/java/ru/magflayer/spectrum/presentation/pages/splash/SplashPresenter.kt
package ru.magflayer.spectrum.presentation.pages.splash

import kotlinx.coroutines.CoroutineExceptionHandler
import kotlinx.coroutines.launch
import moxy.InjectViewState
import ru.magflayer.spectrum.domain.interactor.ColorInfoInteractor
import ru.magflayer.spectrum.presentation.common.mvp.BasePresenter
import javax.inject.Inject

@InjectViewState
class SplashPresenter @Inject constructor(
    private val colorInfoInteractor: ColorInfoInteractor,
) : BasePresenter<SplashView>() {

    internal fun handlePermissionsGranted() {
        val errorHandler = CoroutineExceptionHandler { _, exception ->
            logger.warn("Error while uploaded color info: ", exception)
            viewState.openMainScreen()
        }
        presenterScope.launch(errorHandler) {
            val colorState = colorInfoInteractor.uploadColorInfo()
            logger.debug("Color info loaded: {}", colorState)
            viewState.openMainScreen()
        }
    }
}
